gadget/layout.go

```diff
@@ -336,7 +335,15 @@ func resolveContentPathOrRef(gadgetRootDir, kernelRootDir string, kernelInfo *ke
 		if !ok {
 			return "", false, fmt.Errorf("cannot find %q in kernel info from %q", wantedAsset, kernelRootDir)
 		}
-		if !strutil.ListContains(kernelAsset.Content, wantedContent) {
+		// look for content prefix
+		found := false
+		for _, kcontent := range kernelAsset.Content {
+			if strings.HasPrefix(wantedContent, kcontent) {
```

Comment:
do we have checks to make sure that the filepaths in the kernel.yaml and gadget.yaml are clean? i.e. that we don't have situations like:

```yaml
# kernel.yaml
assets:
  pidtbs:
    update: true
    content:
      - dtbs/////
```

```yaml
# gadget.yaml
 volumes:
  pi:
    schema: mbr
    bootloader: u-boot
    structure:
      - name: ubuntu-seed
        role: system-seed
        filesystem: vfat
        type: 0C
        size: 1200M
        content:
          - source: $kernel:pidtbs/dtbs/broadcom/
            target: /
          - source: $kernel:pidtbs/dtbs/overlays/
            target: /overlays
          - source: boot-assets/
            target: /
```

since with this we clearly don't have a prefix of `"dtbs//////"`. `splitKernelRef` is not doing a filepath.Clean so hence my asking.

Reply:
do we force kcontent to end in / ? we might have strange corner cases and we need to check that either

wantedContent == kcontent or kcontent + '/' (if needed) is a prefix

Reply:
Thanks for these comments. The content paths in kernel.yaml are validated by the "kernel" package to not have more than a single "/". I updated the tests to ensure that kernel.Validate() is run here too so that the gadget tests are actually not able to use incorrect kernel.yaml. 

I also addressed Samueles comment, the code was indeed buggy and did incorrect prefix matching, I added a regression test.